src/audio/copier/copier.c

```diff
@@ -78,7 +80,33 @@ static int copier_init(struct processing_module *mod)
 	 */
 	if (memcpy_s(&cd->config, sizeof(cd->config), copier, sizeof(*copier)) < 0) {
 		ret = -EINVAL;
-		goto error;
+		goto error_cd;
+	}
+
+	/* Allocate memory and store gateway_cfg in runtime. Gateway cfg has to
+	 * be kept even after copier is created e.g. during SET_PIPELINE_STATE
+	 * IPC when dai_config_dma_channel() is called second time and DMA
+	 * config is used to assing dma_channel_id value.
+	 */
+	if (copier->gtw_cfg.config_length) {
+		gtw_cfg_size = (copier->gtw_cfg.config_length << 2);
+		gtw_cfg = rzalloc(SOF_MEM_ZONE_RUNTIME, 0, SOF_MEM_CAPS_RAM,
+				  gtw_cfg_size);
+		if (!gtw_cfg) {
+			ret = -ENOMEM;
+			goto error_cd;
+		}
+
+		cd->gtw_cfg = gtw_cfg;
```

Comment:
I would assign only in success case - below `memcpy_s()` - to avoid a dangling pointer